File: 2022_03_cpp/220325.cpp
//
// Created by z on 3/25/2022.
// # 1029

class Solution {
public:

    static bool cmp(vector<int> a, vector<int> b) {
        return (a[0] - a[1]) < (b[0] - b[1]);
    }

    int twoCitySchedCost(vector<vector<int>> costs) {
        sort(costs.begin(), costs.end(), cmp);
        int ret = 0;
        for (int i = 0; i < costs.size(); i++)
        {
            ret += costs[i][i>=(costs.size()/2)];
        }
        return ret;
    }
};

File: 2022/2022_python/220325.py
#
# created by z on 3/25/2022
# 1029

class Solution:
    def twoCitySchedCost(self, costs: List[List[int]]) -> int:
        costs.sort(key=lambda a: a[0] - a[1])
        ret: int = 0
        half: int = len(costs) / 2
        for i in range(len(costs)):
            ret += costs[i][i >= half]
        return ret